### PR TITLE
Do not allow cray build system patch for later version of otf2

### PR DESCRIPTION
**Pull request by lukebroskop**

### CODE REVIEW

**Comment by hainest** [comment]
@lukebroskop Looks good on Theta. Thanks again!

**Review by hainest — DISMISSED** [review]

**Comment by hainest** [comment]
@tldahlgren @adamjstewart This is ready to be merged at your convenience. Thanks!

File: var/spack/repos/builtin/packages/otf2/package.py [hunk 1]
```diff
@@ -25,21 +25,21 @@ class Otf2(AutotoolsPackage):
...
     version('1.2.1', sha256='1db9fb0789de4a9c3c96042495e4212a22cb581f734a1593813adaf84f2288e4')
 
-    depends_on('autoconf', type='build', when='%cce')
-    depends_on('automake', type='build', when='%cce')
-    depends_on('libtool', type='build', when='%cce')
-    depends_on('m4', type='build', when='%cce')
+    depends_on('autoconf', type='build', when='@2.2 %cce')
+    depends_on('automake', type='build', when='@2.2 %cce')
+    depends_on('libtool', type='build', when='@2.2 %cce')
+    depends_on('m4', type='build', when='@2.2 %cce')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/otf2/package.py]:
```suggestion
    with when('@2.2 %cce'):
        depends_on('autoconf', type='build')
        depends_on('automake', type='build')
        depends_on('libtool', type='build')
        depends_on('m4', type='build')
```

File: var/spack/repos/builtin/packages/otf2/package.py [hunk 2]
```diff
@@ -25,21 +25,21 @@ class Otf2(AutotoolsPackage):
...
 
     # Fix missing initialization of variable resulting in issues when used by
     # APEX/HPX: https://github.com/STEllAR-GROUP/hpx/issues/5239
     patch('collective_callbacks.patch', when='@2.1:2.2')
 
     # when using Cray's cs-prgenv, allow the build system to detect the systems as an XC
-    patch('cray_ac_scorep_sys_detection-m4.patch', when='%cce')
+    patch('cray_ac_scorep_sys_detection-m4.patch', when='@2.2 %cce')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/otf2/package.py:39]: Can you move the patch within the context manager suggested above? Doing so we'll group together all the things needed `when='@2.2 %cce'`).

**Review by alalazo — CHANGES_REQUESTED** [review]
A few minor requests

**Review by alalazo — APPROVED** [review]

**Pull request merged**